src/pages/ProductListing.js

```diff
@@ -1,104 +1,205 @@
-import './productlisting.css'
-import imagebiryani from '../assets/images/biryaniimg.jpeg'
-import imagethali from '../assets/images/thalis.jpeg'
-import imagecake from '../assets/images/cakes.jpeg'
-import imagechinese from '../assets/images/chinese.webp'
-import React,{useState,useEffect} from "react";
-import { apiCall } from '../utility/apiCall'
-import './filter.css'
-import ProuductListingCart from '../components/productlisting-cart'
-import { categoryHandler } from '../utility/filter-functions'
-import { useProductContext } from '../context/product-page-context'
-function ProductListing(){
-    const [productList,setProducutList] = useState([])
-    useEffect(()=>{
-        apiCall('GET','/api/products').then((response)=>{
-       setProducutList( response.data.products)
-      }).catch((e)=>{
-       console.log(e)
+import "./productlisting.css";
+import React, { useState, useContext } from "react";
+import "./filter.css";
+import ProuductListingCart from "../components/productlisting-cart";
+import { categoryHandler } from "../utility/filter-functions";
+import { useProductContext } from "../context/product-page-context";
+import { addToCart } from "../utility/cart-function";
+import { CartContext, CartContextProvider } from "../context/cart-context";
+import { WishListing } from "../components/wishlisting";
+import { useWishList } from "../context/wishlist-context";
+import { addToWishList } from "../utility/wishlist-function";
 
-      })
-   },[])
+function ProductListing() {
+  const {
+    productPageState: { categoryName, productList },
+  } = useProductContext();
 
-  const {productPageState: {categoryName}} = useProductContext();
+  const [range, setRange] = useState(null);
+  const [radio, setRadio] = useState(0);
+  const { cart, setCart } = useContext(CartContext);
+  const { wishList, setWishList } = useWishList();
+  const [checkCategory, setCheckCategory] = useState({
+    veg: false,
+    nonveg: false,
+    gravy: false,
+    dry: false,
+  });
 
+  function radioBasis(list, radio) {
+    if (radio === 0) {
+      return list;
+    }
+    const filteredData = [...list].filter((x) =>
+      Number(x.price < Number(radio))
+    );
+    return filteredData;
+  }
 
-    const prod = categoryHandler(productList, categoryName);
-    console.log(prod, categoryName)
+  function checkBoxFilter(list, category) {
```

Comment:
suggestion: Try to generalize this function. So that it can work even if we increase the categories.